python/lsst/meas/base/classification.py

```diff
@@ -0,0 +1,104 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Definition and registration of classification plugins
+"""
+
+import numpy
+
+import lsst.pex.config
+from .afterburner import AfterburnerPluginConfig, AfterburnerPlugin
+from .pluginRegistry import register
+
+__all__ = (
+    "ABClassificationConfig", "ABClassificationPlugin",
+)
+
+
+class ABClassificationConfig(AfterburnerPluginConfig):
+
+    fluxRatio = lsst.pex.config.Field(dtype=float, default=.925, optional=True,
+                                      doc="critical ratio of model to psf flux")
+    modelErrFactor = lsst.pex.config.Field(dtype=float, default=0.0, optional=True,
+                                           doc="correction factor for modelFlux error")
+    psfErrFactor = lsst.pex.config.Field(dtype=float, default=0.0, optional=True,
+                                         doc="correction factor for psfFlux error")
+
+
+@register("base_ClassificationExtendedness")
+class ABClassificationPlugin(AfterburnerPlugin):
+    """
+    A binary measure of the extendedness of a source, based a simple cut on the ratio of the
+    PSF flux to the model flux.
+
+    Because the fluxes on which this algorithm is based are slot measurements, they can be provided
+    by different algorithms, and the "fluxRatio" threshold used by this algorithm should generally
+    be set differently for different algorithms.  To do this, plot the difference between the PSF
+    magnitude and the model magnitude vs. the PSF magnitude, and look for where the cloud of galaxies
+    begins.
+    """
+
+    ConfigClass = ABClassificationConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    def __init__(self, config, name, schema, metadata):
+        AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.keyProbability = schema.addField(name + "_value", type="D",
+                                              doc="Set to 1 for extended sources, 0 for point sources.")
+        self.keyFlag = schema.addField(name + "_flag", type="Flag", doc="Set to 1 for any fatal failure.")
+
+    def burn(self, measRecord):
+        modelFlux = measRecord.getModelFlux()
+        psfFlux = measRecord.getPsfFlux()
+        modelFluxFlag = (measRecord.getModelFluxFlag()
+                         if measRecord.table.getModelFluxFlagKey().isValid()
+                         else False)
```

Comment:
I don't think it's in the coding standards, but I generally advise putting line breaks after an operator (in this case, `if` and `else` are the operators) rather than before because it makes it clearer that there's something missing if a line is accidentally deleted.